### PR TITLE
Dipolar support for scafacos, various fixes

### PR DESCRIPTION
**Pull request by RudolfWeeber**
This adds dipolar Support for scafacos to the Python interface. It also adds test cases comparing existing dipolar methods against scafacos 2pnfft (from scafacos'  dipoles brnach).

There are also a bunch of Scafacos related fixes + some doc adaptation.

### CODE REVIEW

File: doc/sphinx/inter.rst [hunk 1]
```diff
@@ -1806,36 +1806,27 @@ properties.
...
-solver from scafacos as electrostatics solver for your system, set its
+To use the, e.g.,  ``ewald`` solver from SCAFACOS as electrostatics solver for your system, set its
 cutoff to :math:`1.5` and tune the other parameters for an accuracy of
-:math:`10^{-3}`, use the command
-
-inter coulomb 1.0 scafacos ewald ewald\_r\_cut 1.5 tolerance\_field 1e-3
+:math:`10^{-3}`, use::
+  from espressomd.electrostatics import Scafacos
+  scafacos=Scafacos(bjerrum_length=1,method_name="ewald", 
+    method_params={"ewald\_r\_cut":1.5, "tolerance\_field":1e-3})
```
Comment (hmenke) [inline doc/sphinx/inter.rst]: The backslashes matter here and are printed as is.  They are a relic from the LaTeX docs and should be remove in favour of literal underscores.

File: testsuite/python/scafacos_dipoles_1d_2d.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+from __future__ import print_function
+import espressomd 
+import espressomd.magnetostatics as magnetostatics
+import numpy as np
+import unittest as ut
+
+
+class Scafacos1d2d(ut.TestCase):
+    if "SCAFACOS_DIPOLES" in espressomd.features():
```
Comment (hmenke) [inline testsuite/python/scafacos_dipoles_1d_2d.py]: Please use the `skipIf` decorator as in
````python
@ut.skipIf(not espressomd.has_features(["SCAFACOS_DIPOLES"]),
           "Features not available, skipping test!")
class Scafacos1d2d(ut.TestCase):
````
Saves you also one level of indentation inside the class.

**Review by hmenke — CHANGES_REQUESTED** [review]

**Review by hmenke — APPROVED** [review]

File: src/core/initialize.cpp [hunk 3]
```diff
@@ -67,6 +67,8 @@
 #include "cuda_init.hpp"
 #include "cuda_interface.hpp"
 #include "scafacos.hpp"
+#include "scafacos.hpp"
```
Comment (KaiSzuttor) [inline src/core/initialize.cpp]: duplicate include

File: src/python/espressomd/electrostatics.pyx [hunk 4]
```diff
@@ -23,8 +23,9 @@ from espressomd cimport actors
 from . import actors
 cimport globals
 import numpy as np
-from .scafacos import *
-from . cimport scafacos
+IF SCAFACOS == 1:
+    from .scafacos import *
```
Comment (KaiSzuttor) [inline src/python/espressomd/electrostatics.pyx]: wildcard imports are evil

File: src/python/espressomd/magnetostatics.pyx [hunk 5]
```diff
@@ -19,8 +19,12 @@
 from __future__ import print_function, absolute_import
 include "myconfig.pxi"
 import numpy as np
-from .actors cimport Actor
 from globals cimport temperature
+from .actors cimport *
```
Comment (KaiSzuttor) [inline src/python/espressomd/magnetostatics.pyx]: wildcard imports are evil

File: testsuite/python/coulomb_cloud_wall.py [hunk 6]
```diff
@@ -108,7 +108,7 @@ def test_scafacos_p2nfft(self):
                 self.S.actors.add(Scafacos(bjerrum_length=1, method_name="p2nfft", method_params={
                                   "p2nfft_r_cut": 1.001, "tolerance_field": 1E-4}))
                 self.S.integrator.run(0)
-                self.compare("scafacos_p2nfft", energy=True)
+                self.compare("p3m_gpu", energy=False)
```
Comment (fweik) [inline testsuite/python/coulomb_cloud_wall.py]: This seems wrong to me

Comment (hmenke) [reply testsuite/python/coulomb_cloud_wall.py]: The changed name only leads to a wrong log message.  However, I wonder why energy comparison was disabled.

File: testsuite/python/magnetostaticInteractions.py [hunk 7]
```diff
@@ -21,7 +21,7 @@
 import unittest as ut
 import espressomd
 import numpy as np
-
+import espressomd
```
Comment (hmenke) [inline testsuite/python/magnetostaticInteractions.py:24]: Duplicate import.

**Review by hmenke — APPROVED** [review]

**Pull request merged**